pkg/datastore/datastore.go

```diff
@@ -183,6 +203,25 @@ func (ds *memoryDatastore) GetPackageIDs() string {
 	return strings.Join(keys, ",")
 }
 
+func (ds *memoryDatastore) RegisterOperatorSource(opsrc *v1alpha1.OperatorSource) {
+	ds.rows[opsrc.GetUID()] = &operatorSourceRow{
```

Comment:
This feels more like an addition rather than a registration. `AddOperatorSource` or `InsertOperatorSource` especially given the counter is `RemoveOperatorSource`. Or remove should be called `UnregisterOperatorSource`